rmw_opendds_cpp/src/rmw_publisher.cpp

```diff
@@ -110,20 +111,20 @@ rmw_create_publisher(
   }
   std::string type_name = _create_type_name(callbacks, "msg");
   // Past this point, a failure results in unrolling code in the goto fail block.
-  //DDS::TypeCode * type_code = nullptr;
   DDS::DataWriterQos datawriter_qos;
   DDS::PublisherQos publisher_qos;
   DDS::ReturnCode_t status;
-  DDS::Publisher * dds_publisher = nullptr;
-  DDS::DataWriter * topic_writer = nullptr;
-  DDS::Topic * topic = nullptr;
-  DDS::TopicDescription * topic_description = nullptr;
+  DDS::Publisher_var dds_publisher;
+  DDS::DataWriter_var topic_writer;
+  DDS::Topic_var topic;
+  DDS::TopicDescription_var topic_description;
```

Comment:
Can we not initialize these locals and the point of declaration?

Reply:
Done